### PR TITLE
Indicator Directive

### PR DESCRIPTION
**Pull request by rowa-audil**
Adds the indicator directive into its own package.

Fixes: #210

### CODE REVIEW

File: libs/barista-components/indicator/jest.config.js [hunk 1]
```diff
@@ -0,0 +1,10 @@
+module.exports = {
+  name: 'expandable-text',
```
Comment (ffriedl89) [inline libs/barista-components/indicator/jest.config.js:2]:
```suggestion
  name: 'indicator',
```

File: libs/barista-components/indicator/jest.config.js [hunk 2]
```diff
@@ -0,0 +1,10 @@
...
+  preset: '../../../jest.config.js',
+  coverageDirectory: '../../../coverage/components/expandable-text',
```
Comment (ffriedl89) [inline libs/barista-components/indicator/jest.config.js:4]:
```suggestion
  coverageDirectory: '../../../coverage/components/indicator',
```

File: libs/barista-components/indicator/src/indicator.ts [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+  OnChanges,
+  OnDestroy,
+} from '@angular/core';
+import { Subject } from 'rxjs';
+
+import {
+  CanColor,
+  Constructor,
+  mixinColor,
+} from '../../core/src/common-behaviours/index';
```
Comment (ffriedl89) [inline libs/barista-components/indicator/src/indicator.ts:31]:
```suggestion
} from '@dynatrace/barista-components/core';
```

File: libs/examples/src/indicator/indicator-default-example/indicator-default-example.html [hunk 4]
```diff
@@ -0,0 +1,80 @@
+<dt-tree-table [dataSource]="dataSource" [treeControl]="treeControl">
```
Comment (ffriedl89) [inline libs/examples/src/indicator/indicator-default-example/indicator-default-example.html:1]: I think we should scale down the indicator example to a simple span with a dtIndicator on it and a button to change the color from error to warning. We can reference the more complex example for the tree table in the readme which will showup on barista. Even a link is ok for me there

File: libs/examples/src/indicator/indicator-examples.module.ts [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+
+@NgModule({
+  imports: [
+    DtIndicatorModule,
+    DtTreeTableModule,
+    DtIconModule,
+    DtContextDialogModule,
+    DtInfoGroupModule,
+    DtButtonModule,
+    DtProgressBarModule,
```
Comment (ffriedl89) [inline libs/examples/src/indicator/indicator-examples.module.ts:38]:
```suggestion
    DtButtonModule,
```
Button and Indicator module should be enough for a simple example

File: libs/barista-components/indicator/src/indicator.html [hunk 6]
```diff
@@ -0,0 +1 @@
+<p>dt-indicator works!</p>
```
Comment (tomheller) [inline libs/barista-components/indicator/src/indicator.html:1]:
```suggestion
```

File: libs/examples/src/indicator/indicator-default-example/indicator-default-example.html [hunk 7]
```diff
@@ -0,0 +1,7 @@
+CPU usage:
+<span [dtIndicator]="indicatorState" [dtIndicatorColor]="checkPercentage()"
+  >{{percentage}}%</span
+>
+<br /><br />
+<button dt-button (click)="decreaseValue()">Decrease by 10</button>
+<button dt-button (click)="increaseValue()">Increase by 10</button>
```
Comment (ffriedl89) [inline libs/examples/src/indicator/indicator-default-example/indicator-default-example.html:7]:
```suggestion
<button dt-button (click)="color = color === 'warning' ? 'error' : 'warning'">Toggle color</button>
<button dt-button (click)="shown = !shown">Toggle indicator</button>
```

File: libs/examples/src/indicator/indicator-default-example/indicator-default-example.html [hunk 8]
```diff
@@ -0,0 +1,7 @@
+CPU usage:
+<span [dtIndicator]="indicatorState" [dtIndicatorColor]="checkPercentage()"
```
Comment (ffriedl89) [inline libs/examples/src/indicator/indicator-default-example/indicator-default-example.html:2]:
```suggestion
<span [dtIndicator]="indicatorState" [dtIndicatorColor]="color"
```

File: libs/examples/src/indicator/indicator-default-example/indicator-default-example.ts [hunk 9]
```diff
@@ -0,0 +1,49 @@
...
+@Component({
+  selector: 'dt-example-info-group-default',
+  templateUrl: 'indicator-default-example.html',
+})
+export class DtExampleIndicatorDefault {
+  indicatorState: boolean = true;
+
+  percentage: number = 90;
+
+  color: 'warning' | 'error' | null = null;
```
Comment (ffriedl89) [inline libs/examples/src/indicator/indicator-default-example/indicator-default-example.ts:28]:
```suggestion
  color: 'warning' | 'error' = 'error';
```

File: libs/examples/src/indicator/indicator-default-example/indicator-default-example.ts [hunk 10]
```diff
@@ -0,0 +1,49 @@
...
+import { Component } from '@angular/core';
+
+@Component({
+  selector: 'dt-example-info-group-default',
+  templateUrl: 'indicator-default-example.html',
+})
+export class DtExampleIndicatorDefault {
+  indicatorState: boolean = true;
+
+  percentage: number = 90;
```
Comment (ffriedl89) [inline libs/examples/src/indicator/indicator-default-example/indicator-default-example.ts:26]:
```suggestion
```

File: libs/examples/src/indicator/indicator-default-example/indicator-default-example.ts [hunk 11]
```diff
@@ -0,0 +1,49 @@
...
+  checkPercentage(): 'warning' | 'error' | null {
+    if (this.percentage >= 90 && this.percentage < 100) {
+      return 'warning';
+    } else if (this.percentage >= 100) {
+      return 'error';
+    } else {
+      return null;
+    }
+  }
+}
```
Comment (ffriedl89) [inline libs/examples/src/indicator/indicator-default-example/indicator-default-example.ts:49]: I think we should make the example as simple as possible
```suggestion
  increaseValue(): void {
    this.percentage += 10;
    this.checkPercentage();
  }

  decreaseValue(): void {
    this.percentage -= 10;
    this.checkPercentage();
  }

  checkPercentage(): 'warning' | 'error' | null {
    if (this.percentage >= 90 && this.percentage < 100) {
      return 'warning';
    } else if (this.percentage >= 100) {
      return 'error';
    } else {
      return null;
    }
  }
}
```

File: libs/examples/src/indicator/indicator-default-example/indicator-default-example.ts [hunk 12]
```diff
@@ -0,0 +1,49 @@
...
+ */
+
+import { Component } from '@angular/core';
+
+@Component({
+  selector: 'dt-example-info-group-default',
+  templateUrl: 'indicator-default-example.html',
+})
+export class DtExampleIndicatorDefault {
+  indicatorState: boolean = true;
```
Comment (ffriedl89) [inline libs/examples/src/indicator/indicator-default-example/indicator-default-example.ts:24]:
```suggestion
  shown: boolean = true;
```

File: libs/barista-components/indicator/README.md [hunk 13]
```diff
@@ -6,7 +6,10 @@ warning or an error.
 This directive was introduced to add indicators in the `<dt-table>`, but can be
 used in other components as well to handle error or warning indications.
 
-<ba-live-example name="DtExampleTableProblem" fullwidth></ba-live-example>
+<ba-live-example name="DtExampleIndicatorDefault" fullwidth></ba-live-example>
+
+For a more complex example with the indicator in use visit:
+[Tree Table with DtIndicator](https://barista.dynatrace.com/components/tree-table#dtindicator)
```
Comment (ffriedl89) [inline libs/barista-components/indicator/README.md:12]:
```suggestion
[Tree Table with DtIndicator](/components/tree-table#dtindicator)
```

Comment (ffriedl89) [reply libs/barista-components/indicator/README.md:12]: We can't use the full domain here since this is different on the internal deployment

**Comment by ffriedl89** [comment]
This is a breaking change - please annotate as such in the commit message :)

**Comment by tomheller** [comment]
Looks good, added the blocked label until we release 7.0.0

**Comment by rowa-audil** [comment]
@tomheller In order to remove the deprecated export of the DtIndicatorModule in the Progress Bar module this PR has to be merged. If we are merging this after then I would have to leave the deprecation for later, which means we wouldn't have a clean deprecations removal for 7.0.0.

Why do we have to wait?

**Comment by tomheller** [comment]
If you merge this now, we might not be able to release a 6.x minor version with the remaining features and fixes. 
Please merge the PR when we prepare for the release of 7.0.0. As we are not quite there yet, please hold off.

**Comment by rowa-audil** [comment]
Alright then, my bad I misread the message.

**Pull request merged**